enhancements/oc/mirroring-manifests.md

```diff
@@ -0,0 +1,260 @@
+
+### Pushing to disk while pushing containers to the mirror registry
+
+Users mirroring a release before creating a cluster may want to [push both container images and manifests to disk](#pushing-to-disk).
+But they might also want to push container images directly to a mirror repository (as in [the *applying directly to the target cluster* case](#applying-directly-to-the-target-cluster)) while still pushing manifests to disk (because there is not yet a cluster into which those manifests can be pushed).
+While you could address this use-case with [the *pushing to disk* flow](#pushing-to-disk) followed by an immediate `oc image mirror ...`, it is more convenient to have `oc adm release mirror ...` push to the registry directly.
+Because setting `--to-dir` adjusts the image source, this enhancement extends the command with a new `--manifests-to-dir` that can be used to set a manifest output directory without setting a container image output directory.
+When `--manifests-to-dir` is set, it takes precedence.
+When `--manifests-to-dir` is unset but `--to-dir` is set, the manifest directory is `${TO_DIR}/manifests`, as described in [the previous section](#pushing-to-disk).
+
+### User Stories
+
+#### Release-signature ConfigMaps
+
+Currently a cluster upgrade can be accomplished on a cluster that does not have an active connection to the internet.
+However manual steps are required to create a ConfigMap containing the signature data required for update image verification.
+This enhancement will automatically create the ConfigMap, so the user doesn't have to think about manual steps.
+
+##### Mirroring to a central registry in a fully air gapped environment
+
+A fully air gapped environment is one in which your cluster nodes cannot access the internet.
+For this reason you must mirror the images and manifests to a filesystem disconnected from that environment and then bring that host or removable media across that gap.
+Our documentation refers to this as disconnected mirroring.
+It may also be the case that there are multiple clusters within the air gapped network.
+In such a case it makes sense to configure a central registry in the air gapped environment from which every cluster can pull upgrade imagery.
+Assuming the aforementioned use case and the use of removable media that will be sneakernetted across the gap, the steps for mirroring are:
+
+1. Connect the removable media to a system connected to the internet.
+1. Mirror the images and manifests to a directory on the removable media, using [the `--to-dir` flow](#pushing-manifests-to-disk-for-later-application):
+    ```console
+    $ oc adm release mirror "--to-dir=${REMOVABLE_MEDIA_PATH}/mirror" quay.io/openshift-release-dev/ocp-release:4.3.0-x86_64
+    ```
+    The enhanced `mirror` command will output the path, similar to the example below, to a ConfigMap created containing the image signature.
```

Comment:
I want to see a section in the enhancement that specs all of the files that will be there in the first implementation (signatures, ICSP, ???)

Reply:
> I want to see a section in the enhancement that specs all of the files that will be there in the first implementation (signatures, ICSP, ???)

Do we care?  I think what we put in the Kube manifests directory is basically an implementation detail that does not impact the user experience, and I've made it an explicit non-goal in the current proposal (d327e34230).

Reply:
I care.  You're putting something there.

Reply:
> I care.

We can address you in the oc PR?  This enhancement just tries to pin down the UX, and I don't see why users need to care (and they can look at the self-describing Kube manifests we spit out if they do).